### ISSUE TITLE
Access to my CustoLabel in my CustomCollectionCell to stop collectionView scroll (ATLBaseConversationViewController)

### ISSUE THREAD

**Issue by galu08** [layerhq/Atlas-iOS#857]
Hi, First of all thanks you for this awesome user interface!

Now. I need you help.
I create a “custom payload” by following this tutorial https://developer.layer.com/docs/ios/atlas#custom-payload.

My **CustomCollectionViewCell** has a **CustomLabel**. This CustomLabel declare a protocol  
```Objective-C
@objc protocol CustomLabelDelegate{

    optional func touchesBegan(label:CustomLabel)

    optional func touchesEnded(label:CustomLabel)

 }
```

I want to implement this protocol to by notified when my **CustomLabel** in my **CustomCollectionViewCell**
 was touched. And **STOP** the collectionView scroll. (self.collectionView.scrollEnabled = false )

I imagine that I need to do something like this: 

**SWIFT**
```Swift
class ConversationViewController: ATLConversationViewController {

override func viewDidLoad() {
    super.viewDidLoad()
    self.collectionView.delegate = self	
    …...
}

func collectionView(collectionView: UICollectionView, cellForItemAtIndexPath indexPath: NSIndexPath) -> UICollectionViewCell {

	if let cell = self.collectionView.dequeReusableCell(identifier, index) {

	if let customCell = cell as? CustomCollectionViewCell

		customCell.customLabel.delegate = self   // THIS

		…...

	}	

 }

 func touchesBegan(label:CustomLabel) {

	self.collectionView.scrollEnabled = false

 }
```

You did something like this in ATLConversationViewController.m
```Objective-C
- (UICollectionViewCell *)collectionView:(UICollectionView *)collectionView cellForItemAtIndexPath:(NSIndexPath *)indexPath

{

    LYRMessage *message = [self.conversationDataSource messageAtCollectionViewIndexPath:indexPath];

    NSString *reuseIdentifier = [self reuseIdentifierForMessage:message atIndexPath:indexPath];

    UICollectionViewCell<ATLMessagePresenting> *cell =  [self.collectionView dequeueReusableCellWithReuseIdentifier:reuseIdentifier forIndexPath:indexPath];

    [self configureCell:cell forMessage:message indexPath:indexPath];

    return cell;

}
```
In Conclusion, my customLabel enter in **conflict** with the collectionView scroll. So I want to **disable the collection view scroll when the user touch my custom label**.

Is there a way to access to my CustomLabel in my CustomCollectionViewCell ??

Thanks You !!

Gonza

**Comment by kmahal** [comment]
@galu08 Correct me if I'm wrong, it seems like the issue you mentioned about accessing your label is independent from Atlas itself.  If your label is publicly available in the custom cell then it should be accessible, unless I'm missing something here.

**Comment by galu08** [comment]
@kmahal Thanks for your answer. My CustomLabel is public. My real problem was to get the **cellForItemAtIndexPath**. I believed that I could get it from my ConversationViewController (inherits from ATLConversationViewController). But I couldn't because it is a private method. So, I rewrote **ATLConversationViewController** and made **"UICollectionViewDataSource"** public. Now in ConversationViewController I can get **cellForItemAtIndexPath** (and ask if is my customCell).
very tricky but I didn't found another solution. 
Sorry for my bad English and thanks again!.

**Comment by kmahal** [comment]
@galu08 `collectionView` is a public property which you can access, and you can call `cellForItemAtIndexPath:` on it directly, FYI.  But good to hear you have a solution.  Your English was great, and thank you for writing in!
